OpenRA.Game/Server/Server.cs

```diff
@@ -394,7 +394,7 @@ void ValidateClient(Connection newConn, string data)
 				if (!LobbyInfo.IsSinglePlayer && Map.DefinesUnsafeCustomRules)
 					SendOrderTo(newConn, "Message", "This map contains custom rules. Game experience may change.");
 
-				if (Settings.DisableSinglePlayer)
+				if (!LobbyInfo.GlobalSettings.EnableSingleplayer)
 					SendOrderTo(newConn, "Message", "Singleplayer games have been disabled on this server.");
```

Comment:
Minor scope creep: could you change this message to something like "Two or more human players are required by this server."?  IMO "Singleplayer" has always been a bit too ambiguous.